control-plane/grpc/proto/v1/replica/replica.proto

```diff
@@ -13,8 +13,8 @@ message Replica{
   string name = 2;
   // uuid of the replica
   google.protobuf.StringValue replica_id = 3;
-  // id of the pool
-  string pool_id = 4;
+  // pool on which replica resides
+  Pool pool = 4;
```

Comment:
`PoolRef`? This would need changes are several places, commenting here.

Reply:
PoolRef used inside ReplicaSpec for now. As a modification if we feel to change these as well to PoolRef, will take up as a refactor task.

Reply:
What if we keep this as is (pool_id as a string), and add a new optional uuid separately, would that simplify things?